lib/src/events/guild_events.dart

```diff
@@ -472,3 +477,167 @@ class GuildEventDeleteEvent implements IGuildEventDeleteEvent {
+  /// The guild where this action was executed.
+  Cacheable<Snowflake, IGuild> get guild;
+
+  /// The action which was executed.
+  ActionStructure get action;
+
+  /// The trigger type of rule which was triggered.
+  TriggerTypes get triggerType;
+
+  /// The member which generated the content which triggered the rule.
+  Cacheable<Snowflake, IMember> get member;
+
+  /// The channel in which user content was posted.
+  Cacheable<Snowflake, ITextGuildChannel>? get channel;
+
+  /// The message of any user message which content belongs to.
+  ///
+  /// This will not be present if the message was blocked by automod or the content was not part of the message.
+  Cacheable<Snowflake, IMessage>? get message;
+
+  /// The message id of any system auto moderation messages posted as a result of this action.
+  ///
+  /// `null` if the [action.actionType] is not [ActionTypes.sendAlertMessage].
+  Snowflake? get alertSystemMessage;
+
+  /// The member generated text content.
+  ///
+  /// An empty string if you have not the message content privilegied intent.
+  String get content;
+
+  /// The word or phrase configured in the rule that triggered the rule
+  String? get matchedKeyword;
+
+  /// The substring in content that triggered the rule.
+  ///
+  /// An empty string if you have not the message content privilegied intent.
+  String get matchedContent;
+}
+
+class AutoModeratioActionExecutionEvent extends SnowflakeEntity implements IAutoModerationActionExecutionEvent {
+  @override
+  late final Cacheable<Snowflake, IGuild> guild;
+
+  @override
+  late final ActionStructure action;
+
+  @override
+  late final TriggerTypes triggerType;
+
+  @override
+  late final Cacheable<Snowflake, IMember> member;
+
+  @override
+  late final Cacheable<Snowflake, ITextGuildChannel>? channel;
+
+  @override
+  late final Cacheable<Snowflake, IMessage>? message;
+
+  @override
+  late final Snowflake? alertSystemMessage;
+
+  @override
+  late final String content;
+
+  @override
+  late final String? matchedKeyword;
+
+  @override
+  late final String matchedContent;
+
+  AutoModeratioActionExecutionEvent(RawApiMap rawPayload, INyxx client) : super(Snowflake(rawPayload['d']['rule_id'])) {
+    final raw = rawPayload['d'];
+    guild = GuildCacheable(client, Snowflake(raw['guild_id'] as String));
+    action = ActionStructure(raw['action'] as RawApiMap, client);
+    triggerType = TriggerTypes.fromValue(raw['rule_trigger_type'] as int);
+    member = MemberCacheable(client, Snowflake(raw['user_id'] as Snowflake), guild);
+    channel = raw['channel_id'] != null ? ChannelCacheable(client, Snowflake(raw['channel_id'])) : null;
+    message = raw['message_id'] != null && channel != null ? MessageCacheable(client, Snowflake(raw['message_id']), channel!) : null;
+    alertSystemMessage = raw['alert_system_message_id'] != null ? Snowflake(raw['alert_system_message_id']) : null;
+    content = raw['content'] != null ? raw['content'] as String : '';
```

Comment:
why returning empty string if field is null? I would rather left it null instead of providing default value

Reply:
Discord sends empty strings if message content intent is not passed, however it sends null on auto moderation, so i putted  n empty string for consistency

Reply:
all right